src/node_file.cc

```diff
@@ -1986,8 +1986,26 @@ static void ReadDir(const FunctionCallbackInfo<Value>& args) {
 
   BufferValue path(isolate, args[0]);
   CHECK_NOT_NULL(*path);
+#ifdef _WIN32
+  bool slashCheck = false;
+  if (path.ToStringView().ends_with("/") ||
+      path.ToStringView().ends_with("\\")) {
+    slashCheck = true;
+  }
+#endif
+
   ToNamespacedPath(env, &path);
 
+#ifdef _WIN32
+  if (slashCheck) {
+    std::string result = path.ToString() + "\\";
```

Comment:
maybe this extra string instance could be avoided by appending the `\\` and terminating 0 directly inside `path`.

Reply:
Thank you for the review. Fixed.